pgjdbc/src/test/java/org/postgresql/test/TestUtil.java

```diff
@@ -980,11 +980,43 @@ public static int getBackendPid(Connection conn) throws SQLException {
     return pid;
   }
 
+  public static boolean isPidAlive(Connection conn, int pid) throws SQLException {
+    String sql = "SELECT EXISTS (SELECT * FROM pg_stat_activity WHERE pid = ?) t";
+    try (PreparedStatement stmt = conn.prepareStatement(sql)) {
+      stmt.setInt(1, pid);
+      ResultSet rs = stmt.executeQuery();
+      rs.next();
+      boolean ret = rs.getBoolean(1);
+      rs.close();
+      return ret;
+    }
+  }
+
+  public static boolean waitForBackendTermination(Connection conn, int pid) throws SQLException, InterruptedException {
+    // Default to five-seconds max wait time, with 10ms sleeping
+    return waitForBackendTermination(conn, pid, 5 * 1000L, 10);
```

Comment:
Should the default timeout be like 30 seconds just in case of CI being busy?

Reply:
Sure we can bump that up too. I think I picked 5-seconds because that's what the one place that was actually checking for process termination had.